src/odemis/gui/cont/tabs.py

```diff
@@ -2831,12 +2836,19 @@ def _onSEMpxs(self, pixel_size):
     def _onAlignPos(self, pos):
         """
         Called when the aligner is moved (and the tab is shown)
-        :param pos: (dict str->float) updated position of the aligner
+        :param pos: (dict str->float or None) updated position of the aligner
         """
-        # Return if aligner doesn't have MD_FAV_POS_ACTIVE and MD_FAV_POS_DEACTIVE metadata
+        # Return if aligner doesn't have both MD_FAV_POS_ACTIVE and MD_FAV_POS_DEACTIVE metadata
         md = self._aligner_xy.getMetadata()
         if (model.MD_FAV_POS_ACTIVE and model.MD_FAV_POS_DEACTIVE) not in md:
             return
+        # Move aligner on tab showing to FAV_POS_ACTIVE position (if all axes are referenced)
+        if not self._aligner_activated:
+            if all(self._aligner_xy.referenced.value.values()):
+                self._aligner_xy.moveAbs(md[model.MD_FAV_POS_ACTIVE]).result()  # Wait till movement done
+                self._aligner_activated = True
+            return
+
```

Comment:
Firstly, I don't see why this code should be part of this function. This function is to update the metadata when the aligner is moved. Let's only do that. I think that moving the aligner should be handled by a separate function.

Also, from the experience I recently got from the SPARC alignment tab, I'd recommend the following:
* On show(), if there is any metadata and it's referenced, start the aligner move, and disable the X/Y buttons. Add a callback for the end of the move.
* At the end of the move, enable the X/Y buttons and subscribe the position to _onAlignPos(), if the tab is still shown.

(You need to do it in a callback, because you don't want the GUI to block while the aligner is moving)

You can have a look at odemis.gui.cont.tabs.Sparc2AlignTab, code related to "lens_mover".
You should be able to get rid of `_aligner_activated`.